### ISSUE TITLE
Easier switch between 360° and classic pictures

### ISSUE THREAD

**Issue by PanierAvide** [mistic100/Photo-Sphere-Viewer#682]
Hello,

As I'm making progress on my 360° pictures hosting project (which has a name now : [GeoVisio](https://gitlab.com/PanierAvide/geovisio), [demo here](https://files.pavie.info/clients/geovelo/)), I'm starting to work on a new feature where I may support display of classic (non-360°) pictures.

As now (and as far as I know), PSV offers either :
- Equirectangular Tiles Adapter, which I use as now, but only supports 360° pictures
- Equirectangular Adapter, which offers cropped panorama support, but we can't switch adapters without re-creating the viewer

So to make transitions smoother between 360 and classic pictures, I don't know which option is possible / simpler to implement :
- Add support of cropped panoramas options in Tiles adapter (even if it's only for baseUrl image)
- Allow viewer to switch adapter without destroying it (adapter to choose being an option in VirtualTourPlugin would be :ok_hand: )

Or maybe there is a simpler solution I just missed. Looking forward your thoughts on this.

Best regards :slightly_smiling_face: 

PS : I will present all work done on GeoVisio (and how PSV was helpful !) on the ["State of the Map France"](https://sotm2022.openstreetmap.fr/) OpenStreetMap conference, which will be in Nantes in June 10-12. There will be a dedicated track about open street imagery and "géocommuns" on Friday, if you'd like to attend :wink:

**Comment by mistic100** [comment]
You could probably hack something by adding your own container between the canvas and the navbar, like the markers plugin does.

**Comment by PanierAvide** [comment]
Good to know, I will give it a try and let you know about my progress. If I have something working, would you be interested to have a PR to let it available as a plugin ?

**Comment by PanierAvide** [comment]
Hi, so I made some progress on this, and finally went using directly markers plugin with a "void" panorama (black jpeg tiles) as 360 sphere and an imageLayer marker which is the flat picture to display ([code here](https://gitlab.com/PanierAvide/geovisio/-/blob/feature/flatpic/viewer/src/lib/Viewer.js#L180)).

The rendering is not so bad ([see here](https://files.pavie.info/clients/geovelo/self_hosted/)), however the marker layer comes over the virtual tour arrows. Is there any solution to make arrows shown over marker layer ?

**Comment by mistic100** [comment]
> The rendering is not so bad ([see here](https://files.pavie.info/clients/geovelo/self_hosted/)), however the marker layer comes over the virtual tour arrows. Is there any solution to make arrows shown over marker layer ?

No, the imageLayer is rendered without depth test, so it will always be over any other object in the scene.

**Comment by mistic100** [comment]
As long as it works as intended for standard users yes of course. You can open a PR with the change + update the documentation.
